plugins/flytekit-pydantic/flytekitplugins/pydantic/deserialization.py

```diff
@@ -0,0 +1,134 @@
+import contextlib
+from typing import Any, Callable, Dict, Generator, Iterator, List, Optional, Type, TypeVar, Union
+
+import pydantic
+from flytekit.core import context_manager, type_engine
+from flytekit.models import literals
+
+from flytekit.types import directory, file
+
+from flytekitplugins.pydantic import commons, serialization
+
+# this field is used by pydantic to get the validator method
+PYDANTIC_VALIDATOR_METHOD_NAME = pydantic.BaseModel.__get_validators__.__name__
```

Comment:
Change request (blocking): Could you please account for [this](https://github.com/pydantic/pydantic/blob/f742ca4313fc936a4701f8005b829c78834434f2/docs/migration.md?plain=1#L527) point in the migration guide and make sure that what we build here will work with V2?